cpp/src/arrow/ipc/reader.cc

```diff
@@ -391,12 +342,65 @@ class ArrayLoader {
   ArrayData* out_;
 };
 
+Status DecompressBuffers(const std::vector<std::shared_ptr<ArrayData>>& fields,
+                         Compression::type compression, const IpcReadOptions& options) {
```

Comment:
since you're mutating the contents of fields, should this be
```suggestion
Status DecompressBuffers(Compression::type compression, const IpcReadOptions& options,
                         std::vector<std::shared_ptr<ArrayData>>* fields) {
```
?

Reply:
Sort of a weird case not often seen, since the vector itself is not mutated. When I see `std::vector<T>*` in a function signature that suggests that the vector is modified. Thoughts?

Reply:
It's true; the vector itself isn't mutated and const correctness isn't broken here. I was only thinking of trying to communicate mutation to a future reader. Unfortunately all we have is the ampersand and comments, neither of which is ideal here (... unless we start tossing `std::vector<const T>` around everywhere we want to be clear that the elements are immutable)